### PR TITLE
Add negative pitch clamping

### PR DESCRIPTION
**Pull request by alexschmander**
## Why? What?

This is needed for the pose detection during kick in of #1646 . For the pose detection to work, the referee needs to be fully in frame of the camera. When closer to the referee, the pitch clamping causes the image of the top camera to not fully capture the referee.
Allows for head pitch clamping angles smaller than 0. The negative pitch clamp function is similar to the positive pitch clamping function but offset. 

Fixes #

## How to Test

- CI

### CODE REVIEW

File: crates/control/src/motion/head_motion.rs [hunk 1]
```diff
@@ -163,3 +151,65 @@ impl HeadMotion {
...
+            + interpolation_factor
+                * (head_motion_parameters.inner_maximum_pitch
+                    - head_motion_parameters.outer_maximum_pitch)
+    };
+
+    let minimum_pitch = if controlled_positions.yaw.abs() >= head_motion_parameters.outer_yaw {
+        head_motion_parameters.outer_minimum_pitch
+    } else {
+        let interpolation_factor =
+            0.5 * (PI * controlled_positions.yaw / head_motion_parameters.outer_yaw).cos();
```
Comment (rmburg) [inline crates/control/src/motion/head_motion.rs:174]: Just looking at the code, it looks like there is an `(1.0 + ...)` missing compared to the calculation of `maximum_pitch`. Is this intentional?

Comment (rmburg) [reply crates/control/src/motion/head_motion.rs:174]: It's just a bit confusing since in the code above the interpolation factor (as the name implies) is in the range [0..1], and thus `maximum_pitch` lies inside [`inner_maximum_pitch`..`outer_maximum_pitch`]. Without the `(1.0 + ...)`, `minimum_pitch` is not constrained to the range [`inner_minimum_pitch`..`outer_minimum_pitch`]! It feels like what you're trying to achieve is better done by adjusting the parameters (`inner_minimum_pitch`/`outer_minimum_pitch`)

Comment (ThagonDuarte) [reply crates/control/src/motion/head_motion.rs]: This now needs to be tested again, if the minimum pitch for yaw values where the ear of the Nao is closest to the shoulder is large enough.

Comment (ThagonDuarte) [reply crates/control/src/motion/head_motion.rs]: If that is not the case, the function could easily be scaled along the yaw-axis (x-axis in this graph).

Comment (rmburg) [reply crates/control/src/motion/head_motion.rs]: Nice. Let's test this the next time we have a nao at hand :)

Comment (ThagonDuarte) [reply crates/control/src/motion/head_motion.rs]: We tested with a game pad. Looks beautiful
